ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/notification/NotificationPopup.java

```diff
@@ -0,0 +1,374 @@
+     */
+    private SimplePanel createMessageWidget() {
+        SimplePanel messageWrapper = new SimplePanel();
+        Label messageLabel = new Label();
+        messageWrapper.add(messageLabel);
+        messageWrapper.setStyleName(resources.notificationCss().notificationPopupMessageWrapper());
+        messageWrapper.ensureDebugId(getDebugId(MESSAGE_WRAPPER_DBG_ID, notification));
+        return messageWrapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void onValueChanged() {
+        update();
+    }
+
+    /**
+     * Update notification's widget values.
+     * <p/>
+     * Widget consumes:
+     * <ul>
+     * <li>{@link Notification#title}</li>
+     * <li>{@link Notification#message}</li>
+     * <li>Icon and background color based on {@link StatusNotification.Status}</li>
+     * </ul>
+     */
+    private void update() {
+        Widget titleWidget = titlePanel.getWidget();
+        if (titleWidget != null && titleWidget instanceof Label) {
+            ((Label)titleWidget).setText(notification.getTitle());
+        }
+
+        Widget messageWidget = messagePanel.getWidget();
+        if (messageWidget != null && messageWidget instanceof Label) {
+            ((Label)messageWidget).setText(notification.getMessage());
+        }
+
+        iconPanel.setWidget(getIconBaseOnStatus());
+
+        removeStyleName(resources.notificationCss().notificationStatusProgress());
+        removeStyleName(resources.notificationCss().notificationStatusSuccess());
+        removeStyleName(resources.notificationCss().notificationStatusFail());
+
+        switch (notification.getStatus()) {
+            case PROGRESS:
+                setStyleName(resources.notificationCss().notificationStatusProgress(), true);
+                break;
+            case SUCCESS:
+                setStyleName(resources.notificationCss().notificationStatusSuccess(), true);
+                break;
+            case FAIL:
+                setStyleName(resources.notificationCss().notificationStatusFail(), true);
+                break;
+        }
+
+        if (PROGRESS != notification.getStatus()) {
+            hideTimer.schedule(DEFAULT_TIME);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    protected void onAttach() {
+        boolean isOrWasAttached = isOrWasAttached();
+        super.onAttach();
+
+        if (!isOrWasAttached) {
+            sinkEvents(ONCLICK | ONDBLCLICK | ONMOUSEOUT | ONMOUSEOVER);
+            update();
+
+            new Animation() {
+
+                /** {@inheritDoc} */
+                @Override
+                protected void onUpdate(double progress) {
+                    getElement().getStyle().setOpacity(progress);
+                    getElement().getStyle().setProperty("transform", "scale(" + (2 - progress) + ',' + (2 - progress) + ");");
+                    getElement().getStyle().setProperty("filter", "blur(" + (2 - progress * 2) + "px);");
```

Comment:
I had some troubles with applying pure css animations to popups, so I decided to do this in simple way, via classic animations